### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by glstewart17**
Incorporates authentication with blobstore, however blobstore does not work on deploy.

### CODE REVIEW

**Comment by glstewart17** [comment]
Blobstore now works on the live server!

File: portfolio/src/main/java/com/google/sps/data/Comment.java [hunk 1]
```diff
@@ -14,19 +14,21 @@
...
 
-/** Class containing the author, content, and timestamp of a comment. */
+/** Class containing the author, content, timestamp, and image of a comment. */
 public final class Comment {
 
   private final long id;
   private final String content;
   private final String author;
   private final long timestamp;
+  private final String image;
```
Comment (aditij113) [inline portfolio/src/main/java/com/google/sps/data/Comment.java:24]: Could you add a comment on format - is it the raw pixel byte string or a URL?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -81,16 +100,57 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
...
+    
+    // If user not in datastore, create new user.
+    if (entity == null) {
+      Entity userEntity = new Entity("User", id);
+      userEntity.setProperty("id", id);
+      
+      Random rand = new Random();
+      int number = rand.nextInt(9000) + 1000;
+      String name = "User" + Integer.toString(number);
+      userEntity.setProperty("name", name);
```
Comment (aditij113) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:141]: Is there an easier way to do this? Are the 'ids' unique? In that case could the name be "User"+id ?

File: portfolio/src/main/java/com/google/sps/data/Comment.java (hunk 1 above)
Comment (glstewart17) [reply portfolio/src/main/java/com/google/sps/data/Comment.java:24]: They are all URLs. Just fixed it!

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 2 above)
Comment (glstewart17) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:141]: Right now, uniqueness is not checked/ maintained. Using the id would be much easier though, thanks!

**Pull request merged**